lib/eco_service_web/live/eco_service_live/community_list.ex

```diff
@@ -4,45 +4,73 @@ defmodule EcoServiceWeb.EcoServiceLive.CommunityList do
   alias EcoService.EcoServiceContext
 
   def mount(_params, _session, socket) do
-    params = %{limit: 10, offset: 0}
-    all_communities = EcoServiceContext.fetch_all_communities(params)
+    pag_params = %{limit: 10, offset: 0}
+    all_communities = EcoServiceContext.fetch_all_communities(pag_params)
 
     {:ok,
     socket
     |> assign(:communities, all_communities)
-    |> assign(:limit, params.limit)
-    |> assign(:offset, params.offset)
-    |> assign(:all_records, Enum.count(EcoServiceContext.fetch_all_communities()))
+    |> assign(:limit, pag_params.limit)
+    |> assign(:offset, pag_params.offset)
```

Comment:
You could simply assign `page_params` fully in socket, right? Do you think that'll be more elegant to read and handle? The reason for my suggestion is to avoid repetition.

Reply:
You'd do something like this:
```elixir
|> assign(:page_params, page_params)
```